src/mptcpize.c

```diff
@@ -0,0 +1,226 @@
+// SPDX-License-Identifier: BSD-3-Clause
+#define  _GNU_SOURCE
+
+#include <linux/limits.h>
+
+#include <dlfcn.h>
+#include <errno.h>
+#include <error.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+void help(void)
+{
+	fprintf(stderr, "mptcpize <cmd>\n"
+		"\tavaliable <cmd>:\n"
+		"\trun [-d] prog [<args>]    run target program with specified arguments\n"
+		"\t\t\t forcing MPTCP socket usage instead of TCP\n"
+		"\t\t\t if the '-d' argument is provided, dump messages on stderr\n"
+		"\t\t\t when a TCP socket is forced to MPTCP\n"
+		"\tservice <unit>       create/update <unit>_mptcp systemd unit, which\n"
+		"\t\t\t force the given service to run under the above launcher\n"
+		"\thelp                 show this help\n");
+}
+
+#define MPTCPWRAP_ENV "LD_PRELOAD="PKGLIBDIR"/libmptcpwrap.so.0.0."LIBREVISION
+
+static int run(int argc, char *av[])
+{
+	int i, nr = 0, debug = 0;
+	char **envp, **argv;
+
+	if (argc > 0 && !strcmp(av[0], "-d")) {
+		debug = 1;
+		argc--;
+		av++;
+	}
+
+	if (argc < 1) {
+		fprintf(stderr, "missing command argument");
+		help();
+		return -1;
+	}
+
+	// build environment, copying the current one ...
+	while (environ[nr])
+		nr++;
+	envp = calloc(nr + 3, sizeof(char *));
+	if (!envp)
+		error(1, errno, "can't allocate env list");
+
+	// ... filtering out any 'LD_PRELOAD' ...
+	nr = 0;
+	i = 0;
+	while (environ[nr]) {
+		if (strncmp(environ[nr], "LD_PRELOAD=", 11)) {
```

Comment:
because here we are looking only for the initial 'environ[nr]' sub-string while in other comparison (without the 'n' variant) I'm matching the whole string